src/objects/npc.lua

```diff
@@ -0,0 +1,175 @@
+--[[
+    player.lua
+    basic player object
+--]]
+
+local map = require 'map'
+local obj = require 'obj'
+local sprite = require 'sprite'
+
+return {
+    init = function(self)
+        -- constants
+        self.gravity = self.gravity or 350
+        self.passive_decel = self.passive_decel or 400
+        self.jump_dy = self.jump_dy or -180
+        self.dx_accel = self.dx_accel or 1600
+        self.dx_max = self.dx_max or 150
+        self.moveintent = 2
+        self.jumpintent = 2
+        self.nextmovetimer = 0
+        self.movetimer = 1
+        self.jumptimer = 1
+
+        -- state
+        self.dx = 0
+        self.dy = 0
+        self.jump_enabled = false
+        self.is_walking = false
+        self.direction = 'right'
+
+        -- resources
+        self.idle = sprite.create('32x32_player.png', self.w, self.h, 0.25)
+        self.walk = sprite.create('32x32_player-walk.png', self.w, self.h, 0.1)
+
+        self.spr = self.walk
+        self.spr:play()
+
+    end,
+    update = function(self, dt)
+        -- update the sprite
+        self.spr:update(dt)
+
+        -- find the world geometry layer if we haven't already
+        self.geometry_layer = self.geometry_layer or map.layer_by_name('geometry')
+
+        -- update velocity from inputs
+        self.is_walking = false
+
+        -- assume not walking unless we override it
+        self.spr = self.idle
+
+        -- wait a little bit before deciding what to do next
+        if self.nextmovetimer > 0 then
+            self.nextmovetimer = self.nextmovetimer - (1 * dt)
```

Comment:
This could be written as `self.nextmovetimer - dt` -- or perhaps the `1` could be moved to a constant value in `self`?

Reply:
This has already been changed